albatross/server.py

```diff
@@ -64,41 +108,33 @@ def _parse_header_lines(self, lines):
         return ImmutableCaselessDict(*headers.items())
 
     async def _parse_request(self, request_reader):
-        request_line = await request_reader.readline()
-        request_line = request_line.decode()
-        method, url_string, _ = request_line.split(' ', 2)
-        method = method.upper()
-        url = parse.urlparse(url_string)
+        # Read next event and see if it is a request
 
-        header_lines = []
-        while True:
-            l = await request_reader.readline()
-            if l == b'\r\n':
-                break
-            header_lines.append(l.decode())
+        event = await self.next_event(request_reader)
 
-        headers = self._parse_header_lines(header_lines)
+        if type(event) is h11.Request:
```

Comment:
better to do if not isinstance(event, h11.Request): raise TypeError, and then you dont increase indent on rest of function